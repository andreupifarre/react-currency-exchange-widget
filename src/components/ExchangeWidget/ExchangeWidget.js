import React from 'react';
import Exchanger from '../Exchanger/Exchanger';
import './ExchangeWidget.scss';

const ExchangeWidget = () => (
  <div className="exchange-widget">
    <h2 className="exchange-widget__title">Exchange</h2>
    <Exchanger />
  </div>
);

export default ExchangeWidget;
