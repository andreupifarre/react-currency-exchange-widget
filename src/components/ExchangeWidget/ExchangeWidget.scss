.exchange-widget {
  margin: 0 auto;
  padding: 8px;
  letter-spacing: -0.4px;
  background-color: #f3f4f5;
  box-sizing: border-box;
  max-width: 421px;
  // border: 3px solid #f3f4f5;
  border-radius: 15px;
  display: flex;
  flex-direction: column;
  // box-shadow: 0 1px 2px rgba(0, 0, 0, 0.02), 0 2px 4px rgba(0, 0, 0, 0.02),
  //   0 4px 8px rgba(0, 0, 0, 0.02), 0 8px 16px rgba(0, 0, 0, 0.02),
  //   0 16px 32px rgba(0, 0, 0, 0.07), 0 32px 64px rgba(0, 0, 0, 0.02);

  // box-shadow: 0 32px 64px rgba(0, 0, 0, 0.13);

  // box-shadow: 0 16px 32px rgba(0, 0, 0, 0.05), 0 32px 64px rgba(0, 0, 0, 0.06);

  box-shadow: 0 10px 20px 0 rgba(62, 131, 255, 0.2);

  &__title {
    margin: 0 0 15px;
    font-size: 20px;
    font-weight: 500;
    line-height: 2rem;
  }
}

@media (min-width: 30em) {
  .exchange-widget {
    padding: 34px;
    letter-spacing: 0;
  }
}

@media (min-width: 64em) {
  .exchange-widget {
    width: 480px;
  }
}
