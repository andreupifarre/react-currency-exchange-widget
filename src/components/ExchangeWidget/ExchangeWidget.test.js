import React from 'react';
import { shallow } from 'enzyme';
import ExchangeWidget from './ExchangeWidget';

const wrapper = shallow(<ExchangeWidget />);

describe('ExchangeWidget', () => {
  it('should contain an element with a class .exchange-widget', () => {
    expect(wrapper.find('.exchange-widget').exists()).toBeTruthy();
  });

  it('should contain an H2 element with a class .exchange-widget__title', () => {
    const element = wrapper.find('.exchange-widget__title');
    expect(element.exists()).toBeTruthy();
    expect(element.name()).toEqual('h2');
  });

  it('should contain a Exchanger component', () => {
    expect(wrapper.find('Exchanger').exists()).toBeTruthy();
  });

  it('should render correctly', () => {
    expect(wrapper.getElements()).toMatchSnapshot();
  });
});
