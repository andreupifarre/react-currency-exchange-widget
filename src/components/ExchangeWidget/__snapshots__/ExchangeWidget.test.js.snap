// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeWidget should render correctly 1`] = `
Array [
  <div
    className="exchange-widget"
  >
    <h2
      className="exchange-widget__title"
    >
      Exchange
    </h2>
    <Exchanger />
  </div>,
]
`;
